crates/aptos-crypto/src/ed25519/ed25519_keys.rs

```diff
@@ -137,13 +137,22 @@ impl SigningKey for Ed25519PrivateKey {
     type VerifyingKeyMaterial = Ed25519PublicKey;
     type SignatureMaterial = Ed25519Signature;
 
-    fn sign<T: CryptoHash + Serialize>(&self, message: &T) -> Ed25519Signature {
-        Ed25519PrivateKey::sign_arbitrary_message(self, signing_message(message).as_ref())
+    fn sign<T: CryptoHash + Serialize>(
+        &self,
+        message: &T,
+    ) -> Result<Ed25519Signature, CryptoMaterialError> {
+        Ok(Ed25519PrivateKey::sign_arbitrary_message(
+            self,
+            signing_message(message)?.as_ref(),
+        ))
     }
 
     #[cfg(any(test, feature = "fuzzing"))]
-    fn sign_arbitrary_message(&self, message: &[u8]) -> Ed25519Signature {
-        Ed25519PrivateKey::sign_arbitrary_message(self, message)
+    fn sign_arbitrary_message(
```

Comment:
this shouldn't return Error right? since there's no serialization involved

Reply:
Unfortunately, I think one of the implementations required it

Reply:
Ah nevermind, I see it's calling itself downstream, i can change this back.

Reply:
Also, change it in MultiEd25519. I'll leave a comment there too.